src/main/java/edu/cornell/kfs/sys/service/impl/DocumentMaintenanceServiceImpl.java

```diff
@@ -107,4 +108,12 @@ public void setDocumentMaintenanceDao(DocumentMaintenanceDao documentMaintenance
         this.documentMaintenanceDao = documentMaintenanceDao;
     }
 
+    public void setDocumentRefreshQueue(DocumentRefreshQueue documentRefreshQueue) {
+        this.documentRefreshQueue = documentRefreshQueue;
+    }
+
+    public void setBusinessObjectService(BusinessObjectService businessObjectService) {
+        this.businessObjectService = businessObjectService;
+    }
+

```

Comment:
Shouldn't these two attributes also have getters defined and those getters used instead of directly accessing the attributes?

Reply:
In general, KualiCo's service beans usually have just the setter methods for injecting the required services, and then the injected services are accessed directly instead of going through a getter method. I was aiming to follow that setup here.

There are still some cases in base code that define both getters and setters (or that define protected getters to allow subclass access to private service variables), but they might be less frequent now due to KualiCo's refactoring efforts.

Reply:
I am still not sure I agree with that approach by KualiCo. There have been a number of times we have had to put in a customization to deal with a null pointer exception due a service attribute being null when the class attempts to use it directly. Class org.kuali.kfs.kim.impl.role.RoleServiceBase being just one example. If KualiCo had their standard be to create both class attribute accessors every time, then the amount of customization we would need to do would be encapsulated in the getter when the "null" issues have to be addressed. I will get off my soap box now. :-)

Reply:
In the RoleServiceBase example that you gave, I believe that was a case where the kimTypeInfoService property was null and needed fixing. In that particular case, though, the kimTypeInfoService property was actually one of our customizations to RoleServiceBase, and we had failed to populate the service reference properly in our bean setup. It was a problem we had accidentally introduced, not KualiCo.

The issue with the initialize-if-necessary service getters is that it adds extra clutter to the service classes, and also implies that the service classes might not be fully initialized at startup. With the service injection approach, it allows the service to stipulate that its required services MUST be injected at application startup, and that a null service reference would indicate an initialization problem that should be fixed via the Spring bean setup instead of custom service-loading code.

Reply:
Lets leave things as is for now. Thanks for the clarification.